src/meta.jl

```diff
@@ -0,0 +1,94 @@
+## Functions to serve metadata about the PkgServer itself
+
+function get_pkgserver_version()
+    # Get PkgServer.jl's version and git sha
+    version = Pkg.TOML.parsefile(joinpath(@__DIR__, "..", "Project.toml"))["version"]
+    try
+        repo = LibGit2.GitRepo(dirname(@__DIR__))
+        gitsha = string(LibGit2.GitHash(LibGit2.GitCommit(repo, "HEAD")))
+        return "$(version)-$(gitsha)"
+    catch
+        return "$(version)"
+    end
+end
+# We store this once, since it's not exactly going to change
+const pkgserver_version = get_pkgserver_version()
+
+function get_num_hashnamed_files(dir)
+    # If this directory doesn't exist, then we haven't cached anything!
+    if !isdir(dir)
+        return 0
+    end
+
+    num_files = 0
+    for f in readdir(dir)
+        # only consider things with a name length that is exactly a tree-hash (filters out in-progress downloads, etc...)
+        if length(f) != 40
+            continue
+        end
+
+        # Only consider actual files
+        f_path = joinpath(dir, f)
+        if !isfile(f_path)
+            continue
+        end
+
+        num_files += 1
+    end
+    return num_files
+end
+
+function get_num_pkgs_cached(cache_dir)
+    # If we don't even have a `package` directory, then don't worry about it
+    if !isdir(joinpath(cache_dir, "package"))
+        return 0
+    end
+
+    num_pkgs_cached = 0
+    for d in readdir(joinpath(cache_dir, "package"))
+        # Only consider things with a name that is exactly a UUID (filters out in-progress downloads, etc...)
```

Comment:
We also have a regex for this (`uuid_re`), but that's probably overkill.

Reply:
I'm adding a `response_time` measurement to nginx so that I can see how long this actually takes: I don't want the lock up the pkg server for `10ms` every time someone hits `/meta` just because it's going through parsing a bunch of regexes in a directory that should really be machine-controlled.  ;)

Reply:
We could run a separate process in a loop that just calculates these stats and saves them to a file and then just serve that static file.